documents/int256.go

```
@@ -0,0 +1,108 @@
+package documents
+
+import (
+	"math/big"
+	"regexp"
+
+	"github.com/centrifuge/go-centrifuge/errors"
+	"github.com/centrifuge/go-centrifuge/utils/byteutils"
+	"github.com/ethereum/go-ethereum/common/math"
+)
+
+// signByteIdx is the index of byte that contains the sign bit of the int256 when represented as 32 bytes
+const signByteIdx = 0
+
+// validInt regex for only allowing integers
+var validInt = regexp.MustCompile(`^[-+]?\d+$`)
+
+// Int256 represents a signed 256 bit integer
+type Int256 struct {
+	v big.Int
+}
+
+// NewInt256 creates a new Int256 given a string
+func NewInt256(n string) (*Int256, error) {
+	if !validInt.MatchString(n) {
```

@vedhavyas had to add this because big.NewInt silently ignores decimals